clippy_lints/src/capitalized_acronyms.rs

```diff
@@ -0,0 +1,85 @@
+use crate::utils::span_lint_and_sugg;
+use itertools::Itertools;
+use rustc_ast::ast::{Item, ItemKind, Variant};
+use rustc_errors::Applicability;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::symbol::Ident;
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for camel case name containing a capitalized acronym.
+    ///
+    /// **Why is this bad?** In CamelCase, acronyms count as one word.
+    /// See [naming conventions](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#general-conventions-[rfc-#430])
+    /// for more.
+    ///
+    /// **Known problems:** None.
```

Comment:
I think one problem is, that if two acronyms are chained together they are both lowercased by the suggestion. E.g.:

```
GCCLLVMSomething -> GccllvmSomething
```

Can you also add a test for this, please?